docs/deployment/configuration/performance.rst

```diff
@@ -6,148 +6,211 @@ Optimizing Performance
+     - Number of processes that can work concurrently. Also implies number of workflows that can be executed in parallel. Since FlytePropeller uses ``goroutines``, it can accommodate significantly more processes than the number of physical cores.
+     - ``flyte:propeller:all:free_workers_count``
+     -  A low number may result in higher overall latency for each workflow evaluation round. Larger the number, implies more workflows can be evaluated in parallel. But it should depend on number of CPU cores assigned to FlytePropeller and evaluated against the cost of context switching. A number around 500 - 800 with 4-8 CPU cores usually works fine.
+     - ``plugins.workqueue.config.workers`` Default value: ``10``. 
+   * - Workqueue depth
+     - Current number of workflow IDs in the queue awaiting processing
+     - ``sum(rate(flyte:propeller:all:main_depth[5m]))``
+     - A growing trend indicates the processing queue depth is long and is taking longer to drain, delaying start time for executions.
+     - ``plugins.workqueue.config.maxItems``. Default value: ``10000``
+
+2. Query observed state
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The Kube client config controls the request throughput from FlytePropeller to the Kube API server. These requests may include creating/monitoring Pods or creating/updating FlyteWorkflow CRDs to track workflow execution. 
+The `default configuration provided by K8s <https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/config#GetConfigWithContext>`__ results in very conservative rate-limiting. FlytePropeller provides a default configuration that may offer better performance. 
+However, if your workload involves larger scales (e.g., >5k fanout dynamic or map tasks, >8k concurrent workflows, etc.,) the kube-client rate limiting config may still contribute to a noticeable drop in performance. 
+Increasing the ``qps`` and ``burst`` values may help alleviate back pressure and improve FlytePropeller performance. The default kube-client config applied to Propeller is as follows:
 
 .. code-block:: yaml
 
     propeller:
-        kube-client-config:
-            qps: 100 # Refers to max rate of requests to KubeAPI server
-            burst: 50 # refers to max burst rate to Kube API server
-            timeout: 30s # Refers to timeout when talking with kubeapi server
+      kube-client-config:
+        qps: 100 # Refers to max rate of requests (queries per second) to kube-apiserver
+        burst: 25 # refers to max burst rate. 
```

Comment:
burst should be >= qps. we did a little dive here into the code and realized we have been misconfiguring this for awhile.

Reply:
Uh good finding. I tried to rephrase this section, to the best of my knowledge. I think I'll need to also update the configuration reference page in a separate PR.